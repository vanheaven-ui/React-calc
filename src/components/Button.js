import PropTypes from 'prop-types';

const Button = ({ ButtonName }) => (
  <button type="button">{ButtonName}</button>
);

Button.defaultProps = {
  ButtonName: '--',
};

Button.propTypes = {
  ButtonName: PropTypes.string,
};

export default Button;
