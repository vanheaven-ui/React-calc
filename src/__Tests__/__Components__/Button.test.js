import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react/';
import Button from '../../components/Button';
import '@testing-library/jest-dom';

const handleClick = () => '';

test('renders without crashing', () => {
  const rootNode = document.createElement('div');
  ReactDOM.render(<Button clickHandler={handleClick} />, rootNode);
});

test('component doesn\'t change unexpectedly', () => {
  const btn = renderer.create(
    <Button clickHandler={handleClick}>=</Button>,
  );
  const tree = btn.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders the Button component', () => {
  const { getByRole } = render(
    <Button clickHandler={handleClick} />,
  );
  expect(getByRole('button')).toBeInTheDocument();
});

test('Renders the Button with correct text', () => {
  const { getByRole } = render(
    <Button clickHandler={handleClick} />,
  );
  expect(getByRole('button')).toHaveAttribute('class', '#f5913e default-btn');
});
