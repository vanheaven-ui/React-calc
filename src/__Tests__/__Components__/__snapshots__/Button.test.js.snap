// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component doesn't change unexpectedly 1`] = `
<button
  className="#f5913e default-btn"
  onClick={[Function]}
  type="button"
>
   
</button>
`;
